src/Test/Utilities/Shared/Compilation/OperationTreeVerifier.cs

```diff
@@ -0,0 +1,1065 @@
+            if (hasHeader)
+            {
+                Indent();
+                LogString($"{header}: ");
+            }
+
+            foreach (var operation in operationSequence)
+            {
+                Visit(operation);
+            }
+
+            if (hasHeader)
+            {
+                Unindent();
+            }
+
+            Unindent();
+        }
+
+        public override void VisitVariableDeclarationStatement(IVariableDeclarationStatement operation)
+        {
+            var variablesCountStr = $"{operation.Variables.Length} variables";
+            LogString($"{nameof(IVariableDeclarationStatement)} ({variablesCountStr})");
+            LogCommonPropertiesAndNewLine(operation);
+
+            if (operation.Variables.IsEmpty)
+            {
+                return;
+            }
+
+            VisitOperationSequence(operation.Variables);
+        }
+
+        public override void VisitVariableDeclaration(IVariableDeclaration operation)
+        {
+            LogLocalSymbol(operation.Variable, header: nameof(IVariableDeclaration));
+            LogCommonPropertiesAndNewLine(operation);
+
+            VisitChildOperation(operation.InitialValue, "Initializer");
+        }
+
+        public override void VisitSwitchStatement(ISwitchStatement operation)
+        {
+            var caseCountStr = $"{operation.Cases} cases";
+            LogString($"{nameof(ISwitchStatement)} ({caseCountStr})");
+            LogCommonPropertiesAndNewLine(operation);
+
+            VisitChildOperation(operation.Value, header: "Switch expression");
+            VisitOperationSequence(operation.Cases);
+        }
+
+        public override void VisitSwitchCase(ISwitchCase operation)
+        {
+            var caseClauseCountStr = $"{operation.Clauses} case clauses";
+            var statementCountStr = $"{operation.Body} statements";
+            LogString($"{nameof(ISwitchCase)} ({caseClauseCountStr}, {statementCountStr})");
+            LogCommonPropertiesAndNewLine(operation);
+
+            Indent();
+            LogString("Case clauses: ");
+            VisitOperationSequence(operation.Clauses);
+            LogString("Body: ");
+            VisitOperationSequence(operation.Body);
+            Unindent();
+        }
+
+        public override void VisitWhileUntilLoopStatement(IWhileUntilLoopStatement operation)
+        {
+            LogString(nameof(IForLoopStatement));
+
+            LogString($" (IsTopTest: {operation.IsTopTest}, IsWhile: {operation.IsWhile})");
+            VisitLoopStatementHeader(operation);
+
+            VisitChildOperation(operation.Condition, "Condition");
+            VisitLoopStatementBody(operation);
+        }
+
+        public override void VisitForLoopStatement(IForLoopStatement operation)
+        {
+            LogString(nameof(IForLoopStatement));
+            VisitLoopStatementHeader(operation);
+
+            VisitChildOperation(operation.Condition, "Condition");
```

Comment:
It seems you have basically reimplemented the traversal logic of all IOperations, is it possible to get a similar text output just by calling `base.Visit()` instead? Preferably, `OperationWalker` should be the single source of truth for traversing operation tree.